olmo/checkpoint.py

```
@@ -672,92 +669,144 @@ def save_checkpoint(
-                        if torch.isnan(param).any():
-                            raise ValueError(
-                                f"Module '{module_name}' contains NaNs, this is likely a bug restoring from full checkpoints"
-                            )
+                dist_model.module.load_state_dict(state_dict_to_load, strict=True)
 
             # Load optimizer state.
             if load_optimizer_state:
                 optim_state_dict_to_load = load_state_dict(
                     load_path, "optim.pt", local_cache=local_cache, map_location="cpu"
                 )
-                optim_state_dict_to_load = self._make_optim_state_dict_compatible(
-                    optim_state_dict_to_load,
-                    og_keys_to_new,
-                )
-                gc.collect()
-                torch.cuda.empty_cache()
-                barrier()
-                for turn in range(get_local_world_size()):
-                    log.info("Loading optimizer state turn %d ...", turn)
-                    if turn == get_local_rank():
-                        load_fsdp_optim_state(fsdp_model, optim, optim_state_dict_to_load)
-                        gc.collect()
-                        torch.cuda.empty_cache()
-                    barrier()
-                del optim_state_dict_to_load
+                optim.load_state_dict(optim_state_dict_to_load)
 
-            # Load other state.
-            try:
-                trainer_state = load_state_dict(load_path, "train.pt", local_cache=local_cache)
-            except FileNotFoundError:
-                # for backwards compatibility
-                trainer_state = load_state_dict(load_path, "other.pt", local_cache=local_cache)
+            gc.collect()
+            torch.cuda.empty_cache()
+            barrier()
+        else:
+            log.info(
+                "`FullCheckpointer.restore_checkpoint` only supported for FSDP and DDP distributed strategies!"
+            )
```

This should throw an error